tools/souper-check.cpp

```diff
@@ -134,86 +137,94 @@ int SolveInst(const MemoryBufferRef &MB, Solver *S) {
   int Ret = 0;
   int Success = 0, Fail = 0, Error = 0;
   for (auto Rep : Reps) {
-    if (InferNeg) {
-      bool Negative;
-      if (std::error_code EC = S->negative(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
-                                           Negative, IC)) {
-        llvm::errs() << "Error: " << EC.message() << '\n';
-        Ret = 1;
-        ++Error;
-      } else {
-        llvm::outs() << "known negative from souper: "
-                     << convertToStr(Negative) << "\n";
-        ++Success;
+    if (isInferDFA()) {
+      if (InferNeg) {
+        bool Negative;
+        if (std::error_code EC = S->negative(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
+                                             Negative, IC)) {
+          llvm::errs() << "Error: " << EC.message() << '\n';
+          Ret = 1;
+          ++Error;
+        } else {
+          llvm::outs() << "known negative from souper: "
```

Comment:
please remove "known" from the description of any analysis results besides known bits

Reply:
fixed:)

Reply:
Oops, need to update test cases.